From 3d9ea537749da44d1ad79b95ffa8d319b5af7763 Mon Sep 17 00:00:00 2001
From: Hien Dang <hien.dang.eb@rvc.renesas.com>
Date: Wed, 1 Jun 2016 00:02:01 +0700
Subject: [PATCH 156/909] soc: renesas: rcar_ems_ctrl: Fix cpu weight for
 platform with one targeted CPU

commit 4d8dd2b0561902dbb2b78bebc478c88379b84dfe from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

In current implementation, in case the target_cpus has only one cpu,
the EMS processing will be ignored and warning is raised.

This is bug, because in case system boot with two cores,
so one core can be target of EMS.

This patch fixes above bug.

Signed-off-by: Hien Dang <hien.dang.eb@rvc.renesas.com>
Signed-off-by: Dien Pham <dien.pham.ry@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/renesas/rcar_ems_ctrl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/soc/renesas/rcar_ems_ctrl.c b/drivers/soc/renesas/rcar_ems_ctrl.c
index 9aeb79f6d457..cafca878642f 100644
--- a/drivers/soc/renesas/rcar_ems_ctrl.c
+++ b/drivers/soc/renesas/rcar_ems_ctrl.c
@@ -277,7 +277,7 @@ static int __init rcar_ems_cpu_shutdown_init(void)
 			cpumask_set_cpu(cpu, &freq_scaled_cpus);
 	}
 
-	if (cpumask_weight(&target_cpus) <= 1) {
+	if (cpumask_weight(&target_cpus) == 0) {
 		pr_err("thermal emergency: shutdown cpu none\n");
 		return 0;
 	}
-- 
2.17.1

